testing/unittests/jthreadpooltests.cpp

```diff
@@ -0,0 +1,943 @@
+        // Wait for the running count to be less than 2
+        while (pool->runningCount() >= 2)
+            std::this_thread::sleep_for(std::chrono::milliseconds(50));
+        // Thread8 should start immediately as the pool has capacity
+        params = {lifespan200milliseconds, "Thread9"};
+        testStartNoBlock(&params, "TightlyBoundTestPool", false);
+
+        // Test Thread Pool joinAll
+        //
+        // All threads should complete including Thread8
+        pool->joinAll(true);
+        CPPUNIT_ASSERT_EQUAL(7U, threadStartedCount.load());
+        CPPUNIT_ASSERT_EQUAL(7U, threadCompletedCount.load());
+        CPPUNIT_ASSERT_EQUAL(0U, pool->runningCount());
+        CPPUNIT_ASSERT(pool->running());
+    }
+
+    void testThrottledThreadPoolWithDefaultDelay()
+    {
+        // Create thread pool with max 2 threads and a default start delay of 1 second.
+        // Testing thread start() with and without start delay
+        pool.setown(createThreadPool("ThrottledThreadPoolWithDefaultDelay", factory, true, nullptr, 2));
+
+        // Both threads should start immediately
+        params = {lifespan500milliseconds, "Thread1"};
+        auto beforeStart = std::chrono::high_resolution_clock::now();
+        pool->start(&params, params.name);
+        params = {lifespan500milliseconds, "Thread2"};
+        pool->start(&params, params.name);
+
+        // Wait for both threads to signal that they've started
+        auto duration = measureThreadStartDuration(beforeStart, 2);
+        CPPUNIT_ASSERT(duration.count() < 1000);
+        CPPUNIT_ASSERT_EQUAL(2U, threadStartedCount.load());
+
+        // The Thread3 should not start after waiting on it's start specified timeout as the other
+        // two threads are still running when the timeout expires
+        params = {lifespan500milliseconds, "Thread3"};
+        unsigned thread3ExceptionCount = 0;
+        attemptThreadStartWithTimeout(&params, milliseconds200, "ThrottledThreadPoolWithDefaultDelay", thread3ExceptionCount,
+                                      true, "Start should have thrown an exception as no slots would be available within thread start delay specified");
+        CPPUNIT_ASSERT_EQUAL(2U, threadStartedCount.load());
+
+        // The Thread4 should start after one of the above completes
+        beforeStart = std::chrono::high_resolution_clock::now();
+        params = {lifespan750milliseconds, "Thread4"};
+        unsigned thread4ExceptionCount = 0;
+        attemptThreadStartWithTimeout(&params, milliseconds1000, "ThrottledThreadPoolWithDefaultDelay", thread4ExceptionCount, false);
+        duration = measureThreadStartDuration(beforeStart, 1);
+        CPPUNIT_ASSERT(duration.count() >= 290);
+        CPPUNIT_ASSERT_EQUAL(3U, threadStartedCount.load());
+        CPPUNIT_ASSERT(threadCompletedCount.load() >= 1);
+        CPPUNIT_ASSERT(pool->runningCount() >= 1);
+
+        // Test Thread Pool default start delay
```

Comment:
not sure what this is testing? 
there should be a free slot I think...

maybe comment only misleading/needs clarifying.